addons/cloud_storage_azure/models/ir_attachment.py

```diff
@@ -60,12 +60,14 @@ def get_cloud_storage_azure_user_delegation_key(env):
 
 class IrAttachment(models.Model):
     _inherit = 'ir.attachment'
-    _cloud_storage_azure_url_pattern = re.compile(r'https://(?P<account_name>[\w]+).blob.core.windows.net/(?P<container_name>[\w]+)/(?P<blob_name>[^?]+)')
+    _cloud_storage_azure_url_pattern = re.compile(
+        r'https://(?P<account_name>[a-z\d]{3,24})\.blob\.core\.windows\.net/(?P<container_name>[a-z\d](?:-?[a-z\d]+)+)/(?P<blob_name>[^?]+)',
```

Comment:
Nitpick, this does not need to be too strict.
```suggestion
        r'https://(?P<account_name>[a-z\d]{3,24})\.blob\.core\.windows\.net/(?P<container_name>[a-z\d][a-z\d-]{2,62})/(?P<blob_name>[^?]+)',
```

https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftstorage

Reply:
Yes, the pattern is more used for extraction instead of validation. But it is not bad to have a strict check if the rule is strictly defined. Could you add the link to the comment?

Reply:
Weird, even though that link doesn't mention you cannot end container names with hyphens, it doesn't allow you to create those: 
![image](https://github.com/user-attachments/assets/6b8ac64c-5a94-415f-9ae0-756fec73ddee)

But will update it to fit the documentation.